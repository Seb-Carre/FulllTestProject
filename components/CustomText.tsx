import {StyleSheet, Text, TextProps, TextStyle} from 'react-native';

import React from 'react';

interface CustomTextProps extends TextProps {
  children: any;
  style?: TextStyle | TextStyle[];
  textType?: 'title' | 'regular';
}

/**
 * Extension of Text RN with a configuration, you can use all the properties of Text RN if you need
 * @param {string} text
 */
export default function CustomText({
  style,
  children,
  textType,
  ...props
}: CustomTextProps) {
  let textStyle: object = {};
  // Check the textType is used
  switch (textType) {
    case 'title': {
      textStyle = styles.title;
      break;
    }
    default:
      textStyle = {};
      break;
  }
  // if object style is used to make more styling, we are going to put everything to style our component
  const passedStyles = Array.isArray(style)
    ? Object.assign({}, ...style)
    : style;
  return (
    <Text style={[textStyle, passedStyles && {...passedStyles}]} {...props}>
      {children}
    </Text>
  );
}

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    fontWeight: '600',
  },
});
