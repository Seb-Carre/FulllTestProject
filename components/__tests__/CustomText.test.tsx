import CustomText from '../CustomText';
import React from 'react';
import {render} from '@testing-library/react-native';

it('render the text as snapShot we expected', () => {
  const {toJSON} = render(<CustomText>Hello World</CustomText>);

  expect(toJSON()).toMatchSnapshot();
});

it('renders the customText with the text correctly', () => {
  const text = 'Hello world';
  // Insert text as children
  const {queryByText} = render(<CustomText testID="text">{text}</CustomText>);
  // We check if the text is present
  expect(queryByText(text)).not.toBeNull();
});
