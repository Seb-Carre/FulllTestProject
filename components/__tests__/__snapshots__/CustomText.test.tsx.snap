// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the text as snapShot we expected 1`] = `
<Text
  style={
    [
      {},
      undefined,
    ]
  }
>
  Hello World
</Text>
`;
